servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,416 @@
+    // commit for "table_1" on branch1
+    b1table1 =
+        commitSingleOp(
+            prefix,
+            branch1,
+            b1table1.hash,
+            44,
+            "cid_1",
+            "table_1",
+            "file3",
+            b2table1.content,
+            null);
+    fillExpectedContents(branch1, 1, expectedExpired);
+
+    // commit for "table_1" on branch2
+    b2table1 =
+        commitSingleOp(
+            prefix,
+            branch2,
+            b2table1.hash,
+            45,
+            "cid_1",
+            "table_1",
+            "file4",
+            b1table1.content,
+            null);
+    fillExpectedContents(branch2, 1, expectedLive);
+
+    // commit for "table_1" on branch3
+    CommitOutput b3table1 =
+        commitSingleOp(
+            prefix,
+            branch3,
+            branch3.getHash(),
+            46,
+            "cid_1",
+            "table_1",
+            "file5",
+            b2table1.content,
+            null);
+    fillExpectedContents(branch3, 1, expectedExpired);
+    // delete branch3
+    deleteBranch(branch3.getName(), b3table1.hash);
+
+    // drop table "table_1" on branch1
+    dropTableCommit(prefix, branch1, b1table1.hash, "table_1");
+
+    // one live commit and four expired commits on table_1.
+    performGc(
+        cutoffTime,
+        null,
+        expectedLive,
+        expectedExpired,
+        new HashSet<>(Collections.singletonList(prefix + "_cid_1")));
+  }
+
+  @Test
+  public void testMultiRefCutoffTimeStampPerRef() throws BaseNessieClientServerException {
+    String prefix = "test6";
+    Map<String, Set<Content>> expectedExpired = new HashMap<>();
+    Map<String, Set<Content>> expectedLive = new HashMap<>();
+    Branch branch1 = createBranch("multiRefCutoffTimeStampPerRef_1");
+    Branch branch2 = createBranch("multiRefCutoffTimeStampPerRef_2");
+    // commit for "table_1" on branch1
+    CommitOutput table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    // commit for "table_2" on branch2
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch2, branch2.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    fillExpectedContents(branch1, 1, expectedExpired);
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    Instant defaultCutoffTime = Instant.now();
+
+    // commits for "table_1"
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 45, "cid_1", "table_1", "file4", table1.content, null);
+    fillExpectedContents(branch1, 2, expectedLive);
+
+    // commits for "table_2"
+    table2 =
+        commitSingleOp(
+            prefix, branch2, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    fillExpectedContents(branch2, 1, expectedExpired);
```

Comment:
Why is this expired? It's committed after cut-off-time and branch2 is still alive?

Reply:
this is per reference cutoff time testcase, for branch2 cutoff time is after this commit (line 347)